bundles/org.eclipse.text/src/org/eclipse/jface/text/ListLineTracker.java

```diff
@@ -174,8 +174,10 @@ public final IRegion getLineInformationOfOffset(int position) throws BadLocation
 	public final IRegion getLineInformation(int line) throws BadLocationException {
 		int lines= fLines.size();
 
-		if (line < 0 || line > lines)
-			throw new BadLocationException();
+		if (line > lines)
+			throw new BadLocationException("line > lines: " + line + " > " + lines);  //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
In general I like this as good errors are essential, but the wording is highly confusing here and need one to look at the code what lines / line means.

So I would suggest something like this (also on theother places)
```suggestion
			throw new BadLocationException("The given line "+ line + "  is larger as the number of lines currently tracked (" + lines+")");  //$NON-NLS-1$//$NON-NLS-2$
```

Also using String.format will save you multiple NON_NLS and make it easier to find out how the message will look like.